### PR TITLE
VPN-4933 - Stop DNSPingSender's UDP socket on idle

### PR DESCRIPTION
**Pull request by brizental**
I believe the issue described in VPN-4933 was happening due to the UDP socket going into an error state after the application goes inactive, either by being backgrounded or by putting the device to sleep. That error state plus a bug in Qt's code that leads to an infinite retry loop for the kernal call `sendmsg` was causing the crash.

Going back to the VPN app. In the very specific case of 

1. Turning the VPN on, through the UI
2. Going inactive
3. Turning the VPN off, through the UI

The UDP socket would go into an error state, which could be fixed by a restart of the socket. The error state would cause every call to `writeDatagram` to result in a `NetworkError`. It's important context that `NetworkError` is [the fallback error type for `writeDatagram`](https://codebrowser.dev/qt6/qtbase/src/network/socket/qnativesocketengine_unix.cpp.html#1124). Since a restart addresses it, likely the error was something else.

This error state is not completely unexpected, in [Technical Note TN2277](https://developer.apple.com/library/archive/technotes/tn2277/_index.html) Apple says:

> If the system suspends your app and then, later on, reclaims the resources from underneath your listening socket, your app will no longer be listening for connections, even after it has been resumed. The app may or may not be notified of this, depending on how it manages the listening socket. It's generally easier to avoid this problem entirely by closing the listening socket when the app is in the background.

Instead of restarting the socket when restarting the connection health DNSPingTimer, I thought it made more sense to  instead stop the socket when the timer is off and restart it once it's turned back on.

I also fixed a bug here, that was preventing the connection health to be turned back on when coming back from inactivity.

### CODE REVIEW

**Comment by brizental** [comment]
Note: I have asked Bianca to verify this branch and she is unable to reproduce the bug. Neither am I.

File: src/apps/vpn/dnspingsender.h [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 #include "pingsender.h"
 
+class QThread;
```
Comment (strseb) [inline src/apps/vpn/dnspingsender.h:12]: I think the new thread is unused now, right? :)

**Review by strseb — APPROVED** [review]
r+wc

File: src/apps/vpn/dnspingsender.h [hunk 2]
```diff
@@ -19,11 +19,16 @@ class DnsPingSender final : public PingSender {
...
 
+  void start();
+  void stop() { m_socket.close(); }
+
  private:
   void readData();
 
  private:
   QUdpSocket m_socket;
+  QThread* m_thread = nullptr;
```
[the remark this answers is not in the document]
Comment (brizental) [reply src/apps/vpn/dnspingsender.h:30]: Ack, auto merge messed this up. I'll fix it on the cherry-pick to main.

**Pull request merged**